packages/fx-core/src/plugins/resource/frontend/utils/environment-utils.ts

```diff
@@ -20,4 +22,50 @@ export class EnvironmentUtils {
     }
     await fs.writeFile(envFile, envs);
   }
+
+  static async updateEnvironment(
+    envFilePath: string,
+    runtimeEnv: RuntimeEnvironment,
+    aadEnv: AADEnvironment,
+    functionEnv?: FunctionEnvironment
+  ): Promise<void> {
+    let envs = await EnvironmentUtils.readEnvironments(envFilePath);
+
+    if (!envs) {
+      envs = {};
+    }
+
+    if (functionEnv) {
+      if (!envs[EnvironmentVariables.FuncName]) {
+        envs[EnvironmentVariables.FuncName] = functionEnv.defaultName;
+      }
+
+      if (!envs[EnvironmentVariables.FuncEndpoint]) {
```

Comment:
Nit: this kind of check-update can be wrapped into a function so we don't need these `if`s
e.g.,
```
  const addEnvOnEmpty = (key, v) => {
    if (v !== undefined && env[key] === undefined) {
      env[key] = v;
    }
  }

  addEnvOnEmpty(EnvironmentVariables.FuncName, functionEnv.defaultName);
  addEnvOnEmpty(EnvironmentVariables.FuncEndpoint, functionEnv.endpoint);
  ...
  
```

Reply:
OK, thanks!

Reply:
Updated